ci/profile_runner.py

```
@@ -201,48 +201,56 @@ def _run_with_timeout(
             env["LD_LIBRARY_PATH"] = dll_dir + os.pathsep + env["LD_LIBRARY_PATH"]
         else:
             env["LD_LIBRARY_PATH"] = dll_dir
-        proc = subprocess.Popen(
+        # Raw Popen is deliberate here: the timeout path has to attach a
+        # debugger to a *live* PID before killing it, which neither
+        # subprocess.run nor RunningProcess.run expose. See the drain note
+        # below for why communicate() rather than a poll loop.
+        proc = subprocess.Popen(  # noqa: SRC002 - needs live PID on timeout
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)